src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -51,6 +61,23 @@ def get_observation_histograms(self):
         """
         return self._scheduler.get_observation_histograms()
 
+    def get_expanded_specs(self) -> Dict[str, List[str]]:
+        """Return a dict containing the canonicalized addresses of the specs for this run, and what
+        files they expand to."""
+
+        (addresses,) = self._scheduler.product_request(
+            Addresses, [Params(self._specs, self._options_bootstrapper)]
+        )
+
+        ret = {}
+        for addr in addresses:
+            (targets,) = self._scheduler.product_request(Targets, [Params(Addresses([addr]))])
+            ret[addr.spec] = [
+                tgt.address.filename if tgt.address.is_file_target else str(tgt.address)
```

Comment:
Why normalize to the filename instead of the file address? Note that it's possible to have two distinct file targets referring to the same file, which means that their original BUILD target was different. With your change, we can't disambiguate between which was used.

Reply:
I was originally thinking that the filename would be what an end user wants to see from this feature, rather than the address. In this case, we already know what the original BUILD target was.